crates/metadata-server-grpc/src/grpc.rs

```diff
@@ -8,19 +8,28 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+#[cfg(feature = "grpc-client")]
+use restate_types::net::connect_opts::CommonClientConnectionOptions;
+
 tonic::include_proto!("restate.metadata_server_svc");
 pub const FILE_DESCRIPTOR_SET: &[u8] = tonic::include_file_descriptor_set!("metadata_server_svc");
 
 /// Creates a new MetadataServerSvcClient with appropriate configuration
 #[cfg(feature = "grpc-client")]
-pub fn new_metadata_server_client(
+pub fn new_metadata_server_client<O>(
     channel: tonic::transport::Channel,
-) -> metadata_server_svc_client::MetadataServerSvcClient<tonic::transport::Channel> {
+    connection_options: &O,
+) -> metadata_server_svc_client::MetadataServerSvcClient<tonic::transport::Channel>
+where
+    O: CommonClientConnectionOptions + Send + Sync + ?Sized,
+{
     /// Default send compression for grpc clients
     use tonic::codec::CompressionEncoding;
     pub const DEFAULT_GRPC_COMPRESSION: CompressionEncoding = CompressionEncoding::Zstd;
 
     metadata_server_svc_client::MetadataServerSvcClient::new(channel)
+        .max_decoding_message_size(connection_options.max_decoding_message_size())
+        .max_encoding_message_size(connection_options.max_encoding_message_size())
```

Comment:
nit: Since we only need the `max_decoding_message_size` and `max_encoding_message_size` and not the rest of the exposed methods of `CommonClientConnectionOptions`, maybe it's an option to have a `GrpcClientOptions` trait that is a super trait of the `CommonClientConnectionOptions`. Then one would segregate the interface a little bit cleaner (maybe).

Reply:
Great idea.